flink-metrics/flink-metrics-core/src/main/java/org/apache/flink/metrics/LogicalScopeProvider.java

```diff
@@ -17,13 +17,16 @@
 
 package org.apache.flink.metrics;
 
+import org.apache.flink.annotation.Public;
+
 /**
  * Extension for metric groups that support logical scopes.
  *
  * <p>ATTENTION: This interface is *not* meant for the long-term; it merely removes the need for
  * reporters to depend on flink-runtime in order to access the logical scope. Once the logical scope
  * is properly exposed this interface *will* be removed.
  */
+@Public
```

Comment:
@zentol can you confirm this annotation?

Reply:
yes this should be fine. While we do intend to drop it in 2.0, until then it should remain stable.